src/openzeppelin/tests/mocks/camel721_mock.cairo

```diff
@@ -1,80 +1,103 @@
-#[contract]
+#[starknet::contract]
 mod CamelERC721Mock {
     use starknet::ContractAddress;
     use starknet::get_caller_address;
     use openzeppelin::token::erc721::ERC721;
-    use openzeppelin::utils::serde::SpanSerde;
+    use openzeppelin::token::erc721::ERC721::InternalImpl;
+    use openzeppelin::token::erc721::ERC721::ERC721CamelOnlyImpl;
+
+    #[storage]
+    struct Storage {}
 
     #[constructor]
-    fn constructor(name: felt252, symbol: felt252, tokenId: u256, uri: felt252) {
-        ERC721::initializer(name, symbol);
-        ERC721::_mint(get_caller_address(), tokenId);
-        ERC721::_set_token_uri(tokenId, uri);
+    fn constructor(
+        ref self: ContractState, name: felt252, symbol: felt252, tokenId: u256, uri: felt252
+    ) {
+        let mut unsafe_state = ERC721::unsafe_new_contract_state();
+        InternalImpl::initializer(ref unsafe_state, name, symbol);
+        InternalImpl::_mint(ref unsafe_state, get_caller_address(), tokenId);
+        InternalImpl::_set_token_uri(ref unsafe_state, tokenId, uri);
     }
 
-    // View
-
-    #[view]
-    fn supportsInterface(interfaceId: felt252) -> bool {
-        ERC721::supportsInterface(interfaceId)
+    #[external(v0)]
+    fn supportsInterface(self: @ContractState, interfaceId: felt252) -> bool {
+        let unsafe_state = ERC721::unsafe_new_contract_state();
+        ERC721::SRC5CamelImpl::supportsInterface(@unsafe_state, interfaceId)
     }
 
-    #[view]
-    fn name() -> felt252 {
-        ERC721::name()
+    #[external(v0)]
+    fn name(self: @ContractState) -> felt252 {
+        let unsafe_state = ERC721::unsafe_new_contract_state();
+        ERC721::ERC721MetadataImpl::name(@unsafe_state)
     }
 
-    #[view]
-    fn symbol() -> felt252 {
-        ERC721::symbol()
+    #[external(v0)]
+    fn symbol(self: @ContractState) -> felt252 {
+        let unsafe_state = ERC721::unsafe_new_contract_state();
+        ERC721::ERC721MetadataImpl::symbol(@unsafe_state)
     }
 
-    #[view]
-    fn tokenUri(tokenId: u256) -> felt252 {
-        ERC721::tokenUri(tokenId)
+    #[external(v0)]
+    fn tokenUri(self: @ContractState, tokenId: u256) -> felt252 {
+        let unsafe_state = ERC721::unsafe_new_contract_state();
+        ERC721::ERC721MetadataCamelOnlyImpl::tokenUri(@unsafe_state, tokenId)
     }
 
-    #[view]
-    fn balanceOf(account: ContractAddress) -> u256 {
-        ERC721::balanceOf(account)
+    #[external(v0)]
+    fn balanceOf(self: @ContractState, account: ContractAddress) -> u256 {
+        let unsafe_state = ERC721::unsafe_new_contract_state();
+        ERC721CamelOnlyImpl::balanceOf(@unsafe_state, account)
```

Comment:
I think is easier to rewrite external than to define an interface, and the generic implementation in the mock. Do you think is worth it? We have mocks with only one external method in other modules. Should we write the interface and the generic implementation instead of one external as well?

Reply:
Oh no, I didn't mean define a new interface, I meant reusing the 721 interface (and ISRC5) and just create an impl of the interface inside the mock (to just use one `#[external]`). Just an idea, maybe it's more trouble than it's worth

Reply:
Oh, I got you, I think that makes sense if the interface already contains all the methods. I will update it accordingly.

Reply:
Actually, this doesn't works well with the CamelOnly pattern I've been using, because we don't have the agnostic functions in that trait, and defining the Camel trait just for the mock seems like an overhead to me. Are you ok with continuing like this in the meantime, and maybe refactor this in another PR in the future if we avoid the CamelOnly pattern?

Reply:
Ahh right. Yeah, explicitly setting each method as external is totally fine by me. It could be a while before we can have duplicate methods

Reply:
> Or maybe we can get rid of the agnostic methods in the camel mock? I think we are already doing this in CamelERC20Panic

Your call on this. I thought this would make the mocks a little easier 😅

Reply:
I rather let this for the "Improve Mocks" PR if you agree (same for the directory organization of mocks). I created this issue for that #670